t/t3003-mpi-abort.t

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+#
+
+test_description='Test that MPI_Abort works'
+
+# Notes:
+# We cannot depend on the exit code of the job being the exit code passed
+# to MPI_Abort, since sometimes other tasks are terminated by the shell and
+# exit 143 (the highest exit code becomes the job exit code).  We can depend
+# on the job failing, and the string MPI_Abort appearing in stderr.
+
+. `dirname $0`/sharness.sh
+
+if test -z "$FLUX_TEST_MPI"; then
+    skip_all='skipping MPI tests, FLUX_TEST_MPI not set'
+    test_done
+fi
+
+if ! test -x ${FLUX_BUILD_DIR}/t/mpi/abort; then
+    skip_all='skipping MPI tests, MPI not available/configured'
+    test_done
+fi
+
+# Use convenient sort(1) option to determine if semantic version $1 >= $2
+version_gte() {
+    test "$( (echo $1; echo $2) | sort --version-sort | tail -1)" = $1
+}
+
+#
+# mpich < 3.2.0 simply exits on MPI_Abort() instead of sending the PMI-1
+# abort message.  Don't test this for now because of missing early exit
+# detection in Flux (flux-framework/flux-core#2238)
+#
+mpich_min=3.2.0
+mpich_ver=$(${FLUX_BUILD_DIR}/t/mpi/version | awk '/MPICH Version:/ {print $3}')
+if test -n $mpich_ver && ! version_gte $mpich_ver $mpich_min; then
+    skip_all="skipping MPI abort test on MPICH $mpich_ver (< $mpich_min)"
+    test_done
+fi
+
+export TEST_UNDER_FLUX_CORES_PER_RANK=4
+SIZE=2
+MAX_MPI_SIZE=$(($SIZE*$TEST_UNDER_FLUX_CORES_PER_RANK))
+test_under_flux $SIZE job
+
+OPTS="-ocpu-affinity=off"
+
+diag() {
+	echo "test failed: cat $1"
+	cat $1
+	return 1
+}
+
+# get a PMI server trace in the CI output in case we need it to debug!
+test_expect_success 'MPI_Abort size=1 with PMI server tracing enabled' '
+	${FLUX_BUILD_DIR}/t/mpi/version | head -1 &&
+	! run_timeout 60 flux mini run $OPTS -overbose=2 \
+		${FLUX_BUILD_DIR}/t/mpi/abort 0
```

Comment:
Use of `test_must_fail` is preferred over `!` in sharness tests (it can detect a segfault or other unexpected failure mode and diagnose it as such). In this case there is probably little difference, but it is good to avoid use of `!` so it doesn't get further propagated when tests are copied or used as a reference.

```suggestion
	test_must_fail run_timeout 60 flux mini run $OPTS -overbose=2 \
		${FLUX_BUILD_DIR}/t/mpi/abort 0
```

Same goes for the other uses of `!` below.

Reply:
Oh for some reason I didn't think that would work in combination with `run_timeout` but it does seem to work.  Will fix.